src/populate_result_buffer.jl

```diff
@@ -32,28 +32,17 @@ end
 # type assert in the body of the branch, it is a form of manual dispatch. This is necessary because otherwise,
 # we'd hit dynamic dispatch on custom types which is bad for performance.
 # This code has been adapted from CSV.jl
-@inline function parsecustom!(::Type{customtypes}, row_bytes, pos, len, col_idx, cols, options, _type, row_status, colinds) where {customtypes}
+@inline function parsecustom!(::Type{customtypes}, row_bytes, pos, len, col_idx, cols, options, _type) where {customtypes}
     if @generated
         block = Expr(:block)
-        push!(block.args, quote
-            # TODO: Currently, we shouldn't ever hit this path as we either throw an error for unsupported types
-            # or Parsers throw an error internally for weird custom Integer subtypes that don't have a parse method.
-            row_status |= RowStatus.UnknownTypeError
-            row_status |= RowStatus.HasColumnIndicators
-            colinds = setflag(colinds, col_idx)
-            skip_element!(cols[col_idx])
-            res = Parsers.xparse(String, row_bytes, pos, len, options, Parsers.PosLen31)::Parsers.Result{Parsers.PosLen31}
-            (val, tlen, code) = res.val, res.tlen, res.code
-            return val, tlen, code, row_status, colinds
```

Comment:
This is dead code that was causing type instabilities, because `setflag` doesn't exist anymore -- this made the optimizer give up and made us infer `colinds` as `Any`, causing allocations on when we were marking missing values